lib/iris/fileformats/grib/_load_convert.py

```diff
@@ -758,6 +758,59 @@ def check_range(v1, v2, n, d):
     metadata['dim_coords_and_dims'].append((x_coord, x_dim))
 
 
+def grid_definition_template_40(section, metadata):
+    """
+    Translate template representing a regular Gaussian grid.
+
+    Updates the metadata in-place with the translations.
+
+    Args:
+
+    * section:
+        Dictionary of coded key/value pairs from section 3 of the message.
+
+    * metadata:
+        :class:`collections.OrderedDict` of metadata.
+
+    """
+    scan = scanning_mode(section['scanningMode'])
+
+    major, minor, radius = ellipsoid_geometry(section)
+    cs = ellipsoid(section['shapeOfTheEarth'], major, minor, radius)
+
+    # Calculate longitude points.
+    x_inc = section['iDirectionIncrement'] * _GRID_ACCURACY_IN_DEGREES
+    x_offset = section['longitudeOfFirstGridPoint'] * _GRID_ACCURACY_IN_DEGREES
+    x_direction = -1 if scan.i_negative else 1
+    Ni = section['Ni']
+    x_points = np.arange(Ni, dtype=np.float64) * x_inc * x_direction + x_offset
+
+    # Determine whether the x-points (in degrees) are circular.
+    circular = _is_circular(x_points, 360.0)
+
+    # Get the latitude points.
+    y_points = section['distinctLatitudes']
```

Comment:
The number of latitudes between the equator and pole is enough to define a Gaussian grid, and this is coded in the message (`N`). From this we need to calculate the Gaussian latitudes that correspond to this number.

The calculation just involves defining `2 * N ` Legendre coefficients (trivial, all `0` except the last one which is `1`) and solving for the roots of the polynomial, which is done by first computing the companion matrix and finding the eigenvalues (and a bit of numerical massaging for better estimates of the roots). The roots are directly related to the latitudes by `latitudes = arcsin(roots)`.

The previous step gives us latitudes for the full globe, we'd then have to take a subset based on the encoded first and last latitudes (for a subset of a Gaussian grid) I think, I may be misinterpreting this part.

This would need to be done for every GDT40 message, and with large grid sizes the calculation of latitudes could take a few seconds so caching would probably be important (but easy to do).